src/main/java/com/denizenscript/depenizen/bukkit/properties/breweryx/BreweryPlayerExtensions.java

```diff
@@ -0,0 +1,55 @@
+package com.denizenscript.depenizen.bukkit.properties.breweryx;
+
+import com.denizenscript.denizen.objects.PlayerTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.dre.brewery.BPlayer;
+
+public class BreweryPlayerExtensions {
+
+    public static void register() {
+        // <--[tag]
+        // @attribute <BPlayerTag.drunkenness>
+        // @returns ElementTag
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the drunkness of the brewery player.
+        // -->
+        PlayerTag.tagProcessor.registerTag(ElementTag.class, "drunkenness", (attribute, object) -> {
+            BPlayer bPlayer = BPlayer.hasPlayer(object.getPlayerEntity()) ? BPlayer.get(object.getPlayerEntity()) : null;
+            if (bPlayer != null) {
+                return new ElementTag(bPlayer.getDrunkeness());
+            }
+            return null;
+        });
+
+        // <--[tag]
+        // @attribute <BPlayerTag.quality>
+        // @returns ElementTag
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the quality of the brewery player's drunkenness (drunkeness * drunkeness).
+        // -->
+        PlayerTag.tagProcessor.registerTag(ElementTag.class, "quality", (attribute, object) -> {
```

Comment:
should use registerOnlineOnlyTag if it requires online players

Reply:
Doesn't require them to be online, just for Brewery to have data about them

Reply:
then use `getOfflinePlayer` not `playerEntity`

Reply:
fixed